eventsources/sources/bitbucketserver/start.go

```diff
@@ -191,9 +214,37 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 
 	logger.Info("started processing the Bitbucket Server event source...")
 
+	logger.Info("retrieving the access token credentials...")
+	bitbucketToken, err := common.GetSecretFromVolume(bitbucketserverEventSource.AccessToken)
+	if err != nil {
+		return fmt.Errorf("getting bitbucketserver token. err: %w", err)
+	}
+
+	logger.Info("setting up the client to connect to Bitbucket Server...")
+	bitbucketConfig, err := newBitbucketServerClientCfg(bitbucketserverEventSource)
+	if err != nil {
+		return fmt.Errorf("initializing bitbucketserver client config. err: %w", err)
+	}
+
+	bitbucketURL, err := url.Parse(bitbucketserverEventSource.BitbucketServerBaseURL)
+	if err != nil {
+		return fmt.Errorf("parsing bitbucketserver url. err: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	bitbucketClient := newBitbucketServerClient(ctx, bitbucketConfig, bitbucketToken)
```

Comment:
The `ctx` will be cancelled when this function exits, will the `bitbucketConfig` still be able to used for delete the webhook?

Reply:
Good catch. I created a second client that will be used by `PostInactivate`. I also ensured that the `customClient` uses the `ctx`.

Reply:
In this case, I don't think this is a good approach, maybe just extract the way to get a client to a function?

Reply:
Your instructions aren't clear here. I will take a second look though.

Reply:
Looking at this again, I feel this is the right approach using a separate instance of the bitbucket client for deleting which does not use the canceled context and adding the bitbucket clients to the `Router` struct which can be accessed by the different functions is a common approach to using clients.